pkg/types/builder.go

```diff
@@ -133,7 +133,7 @@ func (g *Builder) AddToBuilder(typeNames *TypeNames, r *resource) (*types.Named,
 
 	for _, p := range r.topLevelRequiredParams {
 		g.validationRules += "\n"
-		g.validationRules += fmt.Sprintf(`// +kubebuilder:validation:XValidation:rule="self.managementPolicy == 'ObserveOnly' || has(self.forProvider.%s)",message="%s is a required parameter"`, p, p)
+		g.validationRules += fmt.Sprintf(`// +kubebuilder:validation:XValidation:rule="!('*' in self.managementPolicies || 'Create' in self.managementPolicies || 'Update' in self.managementPolicies) || has(self.forProvider.%s)",message="%s is a required parameter"`, p, p)
```

Comment:
Are we sure that for a `Delete` operation, we are fine with not setting a required parameter? Did we validate this? I think it's not possible to have delete-only resource right, at least with the default policy sets? Looks like `Delete` policy always comes with the `Create` policy, is this correct? 

If so, are we relying on the default supported policy sets?

Reply:
Aswered a bit [here](https://github.com/upbound/upjet/pull/224#discussion_r1269906549), but now I see that possibly delete wont work if we dont have required fields, not sure how Terraform reacts to that. As I mentioned I didn't test all unsupported combinations. All supported policies have `Delete` in a combination with either `Create` or `Update`.

Reply:
> ... not sure how Terraform reacts to that...
This is also my concern here. I can help validating the behavior. 

Another concern is the validation rules (build-time entities) are not aware of if management policies are actually enabled or not (the decision is only available at runtime). But with these "static" rules, we are conditioning whether, for instance, management policy for a resource is set to `Observe`.

Image a case where the management policy is set to `Observe` and the management policies feature is disabled at runtime. For now, we are relying on the crossplane-runtime to error when the management policies are not enabled but a non-default (non `*`) management policy is declared. I don't think this is a blocker for this PR, just trying to understand the implications so that we can maintain the new feature.